### PR TITLE
Add Travis CI

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This is necessary to ensure changes to formulae in this repo work as expected.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+language: ruby
+os: osx
+env: OSX=10.12
```
Comment (mistydemeo) [inline .travis.yml]: Given that we recommend building formulae in this repo on older versions of OS X, maybe we should test using an older release too.

Comment (MikeMcQuaid) [inline .travis.yml]: @mistydemeo Travis CI has limited numbers of workers running at once per org and a generally limited number of macOS workers. As a result I think it's better sticking with a single version but I'm 👍 to this being an older one. Just wanted to 🚢 this to test the other PRs in progress. Feel free to change in a new PR 👍

Comment (xu-cheng) [inline .travis.yml]: We should just run CI on the oldest available macos image.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  - git -C "$HOMEBREW_REPOSITORY" reset --hard origin/master
+  - brew update || brew update
+  - HOMEBREW_TAP_DIR="$(brew --repo "$TRAVIS_REPO_SLUG")"
+  - mkdir -p "$HOMEBREW_TAP_DIR"
+  - rm -rf "$HOMEBREW_TAP_DIR"
+  - ln -s "$PWD" "$HOMEBREW_TAP_DIR"
+  - export HOMEBREW_DEVELOPER="1"
+  - ulimit -n 1024
+script:
+  - brew test-bot
```
Comment (xu-cheng) [inline .travis.yml]: This should be `brew tests --skip-relocation`

**Pull request merged**